packages/main/src/plugin/extension/extension-development-folders.ts

```diff
@@ -0,0 +1,150 @@
+
+  #apiSender: ApiSenderType;
+
+  #firstLaunch = true;
+
+  #developmentFolders: string[] = [];
+
+  #extensionLoader: ExtensionLoader | undefined;
+
+  // event that will be fired
+  #onDidUpdateDevelopmentFolders: Emitter<ExtensionDevelopmentFolderInfo[]> = new Emitter<
+    ExtensionDevelopmentFolderInfo[]
+  >();
+  onDidUpdateDevelopmentFolders = this.#onDidUpdateDevelopmentFolders.event;
+
+  constructor(configurationRegistry: ConfigurationRegistry, apiSender: ApiSenderType) {
+    this.#configurationRegistry = configurationRegistry;
+    this.#apiSender = apiSender;
+  }
+
+  setExtensionLoader(extensionLoader: ExtensionLoader): void {
+    this.#extensionLoader = extensionLoader;
+  }
+
+  protected refreshFolders(): void {
+    const updatedFolders = this.#configurationRegistry
+      .getConfiguration(ExtensionDevelopmentFolderInfoSettings.SectionName)
+      .get<string[]>(ExtensionDevelopmentFolderInfoSettings.DevelopmentExtensionsFolders, []);
+
+    // first launch ? send the initial value as well
+    // or when value is being changed
+    if (this.#firstLaunch || JSON.stringify(this.#developmentFolders) !== JSON.stringify(updatedFolders)) {
+      this.#developmentFolders = updatedFolders;
+      this.#onDidUpdateDevelopmentFolders.fire(this.getDevelopmentFolders());
+    }
+  }
+
+  init(): void {
+    const developmentExtensionsFoldersConfiguration: IConfigurationNode = {
+      id: 'preferences.extensions',
+      title: 'Extensions',
+      type: 'object',
+      properties: {
+        [`${ExtensionDevelopmentFolderInfoSettings.SectionName}.${ExtensionDevelopmentFolderInfoSettings.DevelopmentExtensionsFolders}`]:
+          {
+            description: 'List of extension folders to monitor',
+            type: 'array',
+            default: [],
+            hidden: true,
+          },
+      },
+    };
+    this.#configurationRegistry.registerConfigurations([developmentExtensionsFoldersConfiguration]);
+
+    // refresh the value when the property is changed
+    this.#configurationRegistry.onDidChangeConfiguration(event => {
+      if (
+        event.key ===
+        `${ExtensionDevelopmentFolderInfoSettings.SectionName}.${ExtensionDevelopmentFolderInfoSettings.DevelopmentExtensionsFolders}`
+      ) {
+        this.refreshFolders();
+      }
+    });
+
+    // initialize and track all development folders
+    this.refreshFolders();
+
+    this.#firstLaunch = false;
+  }
+
+  protected async saveToConfiguration(): Promise<void> {
+    await this.#configurationRegistry
+      .getConfiguration(ExtensionDevelopmentFolderInfoSettings.SectionName)
+      .update(ExtensionDevelopmentFolderInfoSettings.DevelopmentExtensionsFolders, this.#developmentFolders);
+
+    // send an event to refresh the value
+    this.#apiSender.send('extensions-development-folders-changed');
+  }
+
+  async addDevelopmentFolder(path: string): Promise<void> {
+    // check we have extension loader
+    if (!this.#extensionLoader) {
```

Comment:
No way to avoid the circular dependencies ?

Reply:
here, it'll be a follow up to have the analyzer of an extension not being part of extension loader

I don't want to cause a big bang yet

Reply:
Okey np, good to know that this is planned 👍